### ISSUE TITLE
Lighting glitches in S.T.A.L.K.E.R: Shadow of Chernobyl (4500)

### ISSUE THREAD

**Issue by valeth** [ValveSoftware/Proton#197]
There are a few minor lighting glitches in STALKER: SoC.
Those only seem to appear when "Sun Shadow" is turned on under the advanced video settings.
All other graphical settings are set to max.

Here's an example of a glitch
![screenshot_20180823_023311](https://user-images.githubusercontent.com/3198362/44498177-4e533980-a67d-11e8-8315-049681830d02.png)

[System Information](https://gist.github.com/valeth/3231d8ac2e593601015c90d27cd43288#file-steam-system-info-txt)

**Comment by kisak-valve** [comment]
Hello @valeth, please copy your system information from steam (```Steam``` -> ```Help``` -> ```System Information```) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by valeth** [comment]
@kisak-valve updated the description with a link to the system info Gist

**Comment by Barnacle** [comment]
Isn't it a game's engine related bug? There's a couple of console commands like "r2_sun_near_border", that fixes this kind of glitches in Windows. https://www.youtube.com/watch?v=cQ7fyV5XMEs

**Comment by shoober420** [comment]
This is a bug related to sun shadows. It happens on windows and wine as well. As previously mentioned, you can lessen the shadow glitches with console commands, disable sun shadows, or install a mod that updates the graphics (lost alpha).

I personally disabled sun shadows. It gives the game a more dark and gloomy look, and suits the atmosphere more IMO.

**Comment by supashang** [comment]
The frame rate in this game is mental fast! Which is a good thing. It is random from 250 - 430FPS in certain areas! Top right corner of screen Steam FPS counter. However, alt tab makes the game screen go black with no obvious way to recover from.

![screenshot at 2018-08-27 18-25-39](https://user-images.githubusercontent.com/19254047/44674962-0474a580-aa27-11e8-8ee4-dc8758953c25.png)

Description: Ubuntu 18.04.1 LTS
Release:	18.04
Codename:	bionic
Kernel: 4.18.4-041804-generic #201808220230
Full System Info: [https://gist.github.com/supashang/6050de4e5ed8854f58d7152521048170](url)

**Comment by shane-powell** [comment]
This game works well for me using proton. The only in game issues that I encountered were the shadows ones which I fixed by tweaking

 _/home/username/.steam/steam/steamapps/compatdata/4500/pfx/drive_c/documents and settings/all users/documents/stalker-shoc/user.ltx_

All I did was to change the value of r2_sun_near_border to 1.

Specifically I had box shaped shadows in the lower corners of my screen before the above fix.

Do not change user.ltx in the stalker installation directory because that config file appears to be ignored.

As mentioned previously this issue is not a wine one but a game one.

Alt tab works fine for me although I do need to tab out and into the game initially on load to get it to appear. 

Also I do not get a mouse cursor on the steam overlay.

My specs https://gist.github.com/kenjara/b9aae11a5124fca193bbb6c7e5615ee8

**Comment by shane-powell** [comment]
I have played the game for a good number of hours now and just come across this glitch in a thunder storm.

Its strange because its not the first thunder storm and the others looked fine. I am not sure if this is a game glitch or a proton one.

![20180925205000_1](https://user-images.githubusercontent.com/22295415/46040499-64cf3380-c108-11e8-8140-cf7e16cee872.jpg)
![20180925204819_1](https://user-images.githubusercontent.com/22295415/46040512-6bf64180-c108-11e8-9809-07fffaf86c14.jpg)

My specs https://gist.github.com/kenjara/b9aae11a5124fca193bbb6c7e5615ee8

**Comment by shoober420** [comment]
That bug with the thunder storms happened to me when I played with wine-staging 3.10. I’m not sure if this happens on windows though. I almost want to say this happens on windows too.

**Comment by shoober420** [comment]
Here is a video of the lighting glitch during storms I experienced during my play through on Wine. This guy is on Windows Vista, so if this is the same bug you’re experiencing, it’s a bug in the engine. This is Clear Sky, but its the literal exact bug I experienced playing SoC on Wine.

https://m.youtube.com/watch?v=nawa1i79vTM

**Comment by shane-powell** [comment]
Yes that is the similar to what I see. When there is a thunder storm and there are only lightning bolts then its fine. But when its a thunder storm with coloured flashes in the sky that's when things mess up.

**Comment by shoober420** [comment]
I just restarted the game and it went away. A reboot of the whole system may be necessary if that doesn’t work, but this is just a bug in the engine. It’s not related to wine or proton.

The stalker engine is very buggy, but that gives the games their charm IMO. The lightning glitch for example looks really cool.

The shadow glitch mentioned previously isn’t as cool, but it lead to me disabling sun shadows, which in return IMO gave the game a darker and more gloomy atmosphere that suits its environment.

The sun never shines in Chernobyl in my head.

It also gives a nice FPS boost too. Sun shadows look like poop.

**Comment by matou68** [comment]
if i press F6 to quicksave, the game hang, the computer hang, i need to reset.  have others this bug ?.

in the log i have :
4562.022:0008:0036:err:ntdll:RtlpWaitForCriticalSection section 0xc62910 "?" wait timed out in thread 0036, blocked by 0009, retrying (60 sec)

**Comment by shane-powell** [comment]
I have played it for over 10 hours and use F6 often to quicksave. I have not had that error at all so far.

**Comment by matou68** [comment]
some people have problem with save :

https://spcr.netlify.com/app/4500

**Comment by matou68** [comment]
i have upgraded to Ubuntu 18.10 /LLVM 7.0.0 /Mesa 18.2.2 and now i can save with quicksave (F6) without any problem.

**Comment by magist3r** [comment]
I have an issue with alt-tab. After alt-tabbing I can't return to the game. It shows me only black screen with small icon it the top left corner:
![Выделение_074](https://user-images.githubusercontent.com/1011437/57976447-1dc8e200-79fa-11e9-9636-a41d88f26f1a.png)

I have this issue with both Proton 3.16-4 and 4.2-4. I use virtual desktop as a workaround.

I also have this issue with Cuphead (#796).

**Comment by shoober420** [comment]
- Name of the game: S.T.A.L.K.E.R.: Shadow of Chernobyl
- Steam AppID of the game: https://steamdb.info/app/4500/

## System Information
- GPU: GTX 760
- Driver/LLVM version: NVIDIA 440.82
- Kernel version: 5.6.3-zen
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfonvidia
- Proton version: 5.0-6-rc3

LOG:
[steam-4500.zip](https://github.com/ValveSoftware/Proton/files/4452726/steam-4500.zip)

Runs great, I know its whitelisted already, but I thought I would mention one thing. To fix the infamous "Sun Shadows Glitch", enter "r2_sun_near_border 1" in console.

NOTE:
My game did crash during that log, but I feel it was just a fluke.

**Comment by shoober420** [comment]
- Name of the game: S.T.A.L.K.E.R.: Shadow of Chernobyl
- Steam AppID of the game: https://steamdb.info/app/4500/

## System Information
- GPU: Radeon RX 5500 XT
- Driver/LLVM version: Mesa 21.2.0-devel (git-d677120cae) / LLVM 13 git master
- Kernel version: 5.13.0-pf2+
- https://github.com/shoober420/linux-scripts/blob/main/home/shoober420/sysinfoamd
- Proton version: 6.12-GE-1

LOG:
[steam-4500.log](https://github.com/ValveSoftware/Proton/files/6788257/steam-4500.log)

The game crashes on launch.

**Comment by kisak-valve** [comment]
Hello @shoober420, since this game is on the whitelist, we want issues with this game reported separately, one issue per issue report.

Please open a new issue report. Additionally, please reproduce your issue with a vanilla version of Proton.
